Set "All Time " range for query log datepicker based on Database ranges (luxon)

PromoFaux opened the pull request:
### **What does this PR aim to accomplish?:**

Builds upon #3655 and uses the newly published `earliest_timestamp_disk` to set the earliest possible date in the daterangepicker.

Due to an issue in the new daterangepicker library, there is a cosmetic issue which is fixed with a locally applied patch temporarily. This can be removed if/when the vendor integrates the upstream PR https://github.com/Wernfried/daterangepicker/pull/1

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

PromoFaux committed:
Refactor date range initialization to fetch earliest timestamp from API and update date picker accordingly.

We use the on-disk earliest timestamp to get the complete range.

Signed-off-by: Adam Warner <me@adamwarner.co.uk>

yubiuser commented on scripts/js/queries.js:
```diff
@@ -40,39 +42,60 @@ function getDnssecConfig() {
+    // earliest_timestamp is provided in seconds since epoch
+    beginningOfTime = Number(data.earliest_timestamp_disk);
+    // Round down to nearest 5-minute segment (300 seconds)
+    beginningOfTime = Math.floor(beginningOfTime / 300) * 300;
```
Could also work with `luxon` `toRelative` using the rounding and padding options

https://moment.github.io/luxon/api-docs/index.html#datetimetorelative

yubiuser commented on scripts/js/queries.js:
```diff
@@ -9,10 +9,12 @@
+// These values are provided by the API (/info/database).
+// We initialize them as null and populate them during page init.
+let beginningOfTime = null; // seconds since epoch (set from API: info/database.earliest_timestamp)
+let endOfTime = null; // seconds since epoch (set to end of today)
```
```suggestion
let endOfTime = null; // seconds since epoch (set to start of tomorrow)
```

PromoFaux replied:
It's also never written to outside of the init function. Might as well `const` and init it here.

yubiuser commented on scripts/js/queries.js:
```diff
@@ -40,39 +42,60 @@ function getDnssecConfig() {
+        Yesterday: [now.minus({ days: 1 }).startOf("day"), now.minus({ days: 1 }).endOf("day")],
+        "Last 7 Days": [now.minus({ days: 6 }), maxDateDt],
+        "Last 30 Days": [now.minus({ days: 29 }), maxDateDt],
+        "This Month": [now.startOf("month"), now.endOf("month")],
```
```suggestion
        "This Month": [now.startOf("month"), maxDateDt],
```

yubiuser commented on scripts/js/queries.js:
```diff
@@ -40,39 +42,60 @@ function getDnssecConfig() {
-          luxon.DateTime.fromMillis(beginningOfTime * 1000),
-          luxon.DateTime.fromMillis(endOfTime * 1000),
-        ], // convert to milliseconds since epoch
+        "This Year": [now.startOf("year"), now.endOf("year")],
```
```suggestion
        "This Year": [now.startOf("year"), maxDateDt],
```

yubiuser reviewed (commented):
<img width="1394" height="420" alt="Image" src="https://github.com/user-attachments/assets/0bf4c09e-e051-4f51-bf30-5157545d398c" />

yubiuser commented:
Unrelated to your changes: the end month selector does not stick
![Peek 2025-11-10 08-51](https://github.com/user-attachments/assets/9b7659ca-4daa-41ef-b214-792b4ac8d181)

rdwebdesign commented:
I also found a strange issue:

If you select a time range like "This month" and then you select "Custom Range", the selection should show the current month with all days selected until Today, but this is what I see:

![strange_selection](https://github.com/user-attachments/assets/e9e7d491-9f23-4969-b2c2-1416a2b15ffe)

rdwebdesign commented on scripts/js/queries.js:
```diff
@@ -40,39 +42,60 @@ function getDnssecConfig() {
       timePicker: true,
       timePickerIncrement: 5,
       timePicker24Hour: true,
       locale: { format: dateformat },
```
If we want to keep using "Sunday" as the first day, we need to add `firstDay` option:

```suggestion
      locale: { 
        format: dateformat,
        firstDay: 7,
      },
```

PromoFaux committed:
Treat 0.0 response as NULL. Also get the in-memory timetamp, and then use whichever of the two timestamps is smallest (and non-zero)

Signed-off-by: Adam Warner <me@adamwarner.co.uk>

rdwebdesign commented on scripts/js/queries.js:
```diff
@@ -40,39 +42,85 @@ function getDnssecConfig() {
+    // Round down to nearest 5-minute segment (300 seconds) if valid
+    if (beginningOfTime !== null) {
+      beginningOfTime = Math.floor(beginningOfTime / 300) * 300;
+    }
```
Fixing linter errors:

```suggestion
    const diskTimestamp = Number(data.earliest_timestamp_disk);
    const  memoryTimestamp = Number(data.earliest_timestamp);

    // Filter out zero/invalid timestamps
    const validTimestamps = [diskTimestamp, memoryTimestamp].filter(ts => ts > 0);

    // Use the smallest valid timestamp, or null if none exist
    beginningOfTime = validTimestamps.length > 0 ? Math.min(...validTimestamps) : null;

    // Round down to nearest 5-minute segment (300 seconds) if valid
    if (beginningOfTime !== null) {
      beginningOfTime = Math.floor(beginningOfTime / 300) * 300;
    }
```

yubiuser commented:
Two reported issues have been fixed upstream. I'll update the vendor files in my branch

PromoFaux committed:
Reviewer suggestions

Signed-off-by: Adam Warner <me@adamwarner.co.uk>

yubiuser commented on scripts/js/queries.js:
```diff
@@ -9,10 +9,14 @@
+// We initialize them as null and populate them during page init.
+let beginningOfTime = null; // seconds since epoch (set from API: info/database.earliest_timestamp)
+// endOfTime should be the start of tomorrow in seconds since epoch
+// We don't use 23:59:59 as the picker increments are set to 5 minutes
```
```suggestion
// We don't use 23:59:59 (endOf("day")) as the picker increments are set to 5 minutes
```

yubiuser commented on scripts/js/queries.js:
```diff
@@ -40,39 +44,84 @@ function getDnssecConfig() {
+  if (beginningOfTime === null) {
+    $("#querytime").prop("disabled", true);
+    $("#querytime").addClass("disabled");
+    $("#querytime-note").text("No data in the database");
```
```suggestion
    $("#querytime-note").text("ℹ️ No data in the database");
```

yubiuser commented on scripts/js/queries.js:
```diff
@@ -40,39 +44,84 @@ function getDnssecConfig() {
-      endDate: luxon.DateTime.fromMillis(until * 1000), // convert to milliseconds since epoch
+      locale: {
+        format: dateformat,
+        firstDay: 7,
```
This might be a British thing, but my week starts on Monday
```suggestion
        firstDay: 1,
```

rdwebdesign replied:
I think this is more a personal choice.

Here, almost every calendar shows Sunday as the first column and starting a calendar with Monday is strange.

yubiuser replied:
We could implement something based on the local used by user's system
https://moment.github.io/luxon/#/intl?id=locale-based-weeks

yubiuser commented on scripts/js/queries.js:
```diff
@@ -40,39 +44,84 @@ function getDnssecConfig() {
+        "Last Hour": [now.minus({ hours: 1 }), now],
+        Today: [now.startOf("day"), maxDateDt],
+        Yesterday: [now.minus({ days: 1 }).startOf("day"), now.minus({ days: 1 }).endOf("day")],
+        "Last 7 Days": [now.minus({ days: 6 }), maxDateDt],
```
```suggestion
        "Last 7 Days": [now.minus({ days: 6 }).startOf("day"), maxDateDt],
```

yubiuser commented on scripts/js/queries.js:
```diff
@@ -40,39 +44,84 @@ function getDnssecConfig() {
+        Today: [now.startOf("day"), maxDateDt],
+        Yesterday: [now.minus({ days: 1 }).startOf("day"), now.minus({ days: 1 }).endOf("day")],
+        "Last 7 Days": [now.minus({ days: 6 }), maxDateDt],
+        "Last 30 Days": [now.minus({ days: 29 }), maxDateDt],
```
```suggestion
        "Last 30 Days": [now.minus({ days: 29 }).startOf("day"), maxDateDt],
```

yubiuser commented:
@PromoFaux can we proceed with this branch? I'd like to get it merged, so we only have one luxon branch we need to keep adjusted with `development`

PromoFaux committed:
More reviewer suggestions

Signed-off-by: Adam Warner <me@adamwarner.co.uk>

PromoFaux commented:
<strike> @yubiuser I rebased on development - so perhaps we can just replace your branch with this one? </strike>

Better still, I rebased your branch on development and then rebased this branch on your branch.

PromoFaux commented:
Added commits directly to `luxon` branch

yubiuser commented:
@rdwebdesign Did you report your issue to the upstream repo yet?

rdwebdesign commented:
No. I forgot.

Actually, before re-reading this issue I didn't even remember what was exactly the issue.
I will test again to get more details about the issue, when I have the time.
